java-checks/src/main/java/org/sonar/java/checks/PrintfCheck.java

```diff
@@ -115,6 +115,23 @@ protected void onMethodInvocationFound(MethodInvocationTree mit) {
     }
   }
 
+  private static Set<Integer> argIndexes(List<String> params) {
+    int index = 0;
+    Set<Integer> result = Sets.newHashSet();
+    for (String rawParam : params) {
+      if (rawParam.contains("$")) {
+        result.add(getIndex(rawParam));
+      } else if (!rawParam.startsWith("<")) {
+        result.add(index++);
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Extract this increment or decrement operator into a dedicated statement. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.internal.sonarsource.com/coding_rules#rule_key=squid%3AS881)